aspnetcore/security/authentication/configure-jwt-bearer-authentication.md

```diff
@@ -0,0 +1,365 @@
+This is a great way to implement this requirement but is complicated if you must implement the OAuth flow yourself. 
+
+See [OAuth 2.0 Token Exchange](https://datatracker.ietf.org/doc/html/rfc8693)
+
+### Use Microsoft on behalf of flow to request a new delegated access token
+This works very well and is super easy to implement when using Microsoft .Identity.Web. It only works with Microsoft Entra ID, Microsoft Entra External ID and Microsoft Azure AD B2C.
+
+See [[Microsoft identity platform and OAuth 2.0 On-Behalf-Of flow](/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow)
+
+### Use the same delegated access token sent to the API
+This is easy to implement but the access token has access to both APIs. Yarp reverse proxy can be used to implement this. This is easy to implement.
+
+### Use OAuth client credentials flow and use an application access token
+This is easy to implement but the client application has full application access and not a delegated access token. The token should be cached in the client API application.
+
+> Note
+> Any app-to-app security would also work. Certificate authentication can be used or in Azure a managed identity can be used. 
+
+## Handling access tokens
+
+When using access tokens in a client application, the access tokens need to be rotated, persisted and stored somewhere on the server. In a web application, cookies are used to secure the session. The cookies should not contain the access tokens as the size of the cookie would become too large.
+
+Duende.AccessTokenManagement.OpenIdConnect is a great Nuget package for handling and managing access tokens in the client application. 
+
+> Note
+> If deploying the production, the cache should work in a mutli-instance deployment and a persistent cache is normally required.
+
+Some secure token servers encrypt the access tokens. Access tokens do not require any format. When using OAuth introspection, a reference token is used instead of an access token. A client (UI) application should never open an access token as the access token is not intended for this. Only an API for which the access token was created for should open the access token. 
+
+* Do not open access tokens in a UI application
+* Do not send the ID token to the APIs
+* Access tokens can have any format
+* Access tokens can be encrypted
+* Access tokens expire and need to be rotated
+* Access tokens are persisted on a secure backend server
+
+## YARP
+
+YARP (Yet Another Reverse Proxy) is a great tool for handling http requests and forwarding the requests to other APIs. YARP can implement security logic for acquiring new access credentials. YARP is used a lot in the backend for frontend (BFF) security architecture.
+
+## Testing APIs
+
+Testing secure APIs can be implemented in different ways. Integration tests and containers with access tokens can be used to test the APIs. You can also create the access tokens using the dotnet user-jwts tool.
```

Comment:
```suggestion
Integration tests and containers with access tokens can be used to test secure APIs. Access tokens can be created using the [dotnet user-jwts tool](/aspnet/core/security/authentication/jwt-authn).
```

Reply:
@damienbod @halter73  I fixed the link and moved it to where it's used. It reads better and the MS style guide mantra is less is more.

<img width="339" alt="image" src="https://github.com/user-attachments/assets/02712651-85d5-4518-915b-b9082baba0d7" />